### PR TITLE
[Chore] Implement Flask Mail

### PR DESCRIPTION
**Pull request by Andretalik**
## Description (what problem you're fixing)

  - The Previous implementation of the notifications was buggy and inconsistent.

## Fix (what you did to fix it)

  - Replaced the implementation with FlaskMail
  - Added an endpoint to test the notifications remotely.

## How to test (describe how to test your PR)

- Clone the repository, install docker, run `make test` within the repository.
- Trigger a build on CircleCI.

### CODE REVIEW

File: src/api/endpoints/healthcheck_email.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        """Trigger send of email to predetermined email address."""
+        payload = request.get_json(silent=True)
+        if not payload:
+            return response_builder(dict(
+                                    message="Data for sending must "
+                                            "be provided.",
+                                    status="fail",
+                                    ), 400)
+
+        recipient = payload.get("recipient")
```
Comment (LarryWachira) [inline src/api/endpoints/healthcheck_email.py]: Maybe a marshmallow email validation here?

Comment (Andretalik) [reply src/api/endpoints/healthcheck_email.py]: On it.

File: src/requirements.txt [hunk 2]
```diff
@@ -43,6 +47,7 @@ psycopg2==2.7.3.1
 ptyprocess==0.5.2
 py==1.4.34
 pycodestyle==2.3.1
+pycrypto==2.6.1
```
Comment (LarryWachira) [inline src/requirements.txt]: This and pycrptodome not used anywhere last we checked.

Comment (Andretalik) [reply src/requirements.txt]: I'll remove it then

File: src/api/endpoints/healthcheck_email.py (hunk 1 above)
Comment (LarryWachira) [reply src/api/endpoints/healthcheck_email.py]: It's registered with the app with no restrictions on who can use it. Essentially all Andelans can use it. Imagine an Andelan with ill intent.

File: src/api/utils/notifications/email_notices.py [hunk 3]
```diff
@@ -1,7 +1,11 @@
 """Notifications module."""
 import os
+import smtplib
```
Comment (LarryWachira) [inline src/api/utils/notifications/email_notices.py]: How do you use this?

File: src/api/endpoints/healthcheck_email.py (hunk 1 above)
Comment (Andretalik) [reply src/api/endpoints/healthcheck_email.py]: makes sense

**Comment by kimani-ndegwa** [comment]
@Andretalik Please fix these conflicts

File: src/api/endpoints/healthcheck_email/__init__.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+    email_healthcheck_service = Blueprint('email_healthcheck', __name__)
+    email_healthcheck_api = Api(email_healthcheck_service)
+
+    # activities endpoints
+    email_healthcheck_api.add_resource(
+        NotificationsAPI, "/api/v1/health/email",
+        "/api/v1/health/email/",
+        endpoint="email_healthcheck",
+        resource_class_kwargs={
```
Comment (LarryWachira) [inline src/api/endpoints/healthcheck_email/__init__.py]: You don't need this particular kwarg.

File: src/requirements.txt [hunk 5]
```diff
@@ -13,13 +15,16 @@ coveralls==1.2.0
 decorator==4.3.0
 docopt==0.6.2
 ecdsa==0.13
-Flask==0.12.4
```
Comment (LarryWachira) [inline src/requirements.txt]: This needs to update. There's a vulnerability in 0.12.2

File: src/api/endpoints/healthcheck_email/healthcheck_email.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
+"""Module to test Notifications being sent out."""
+# system imports
+from flask import request, current_app
+from flask_restplus import Resource
```
Comment (georgreen) [inline src/api/endpoints/healthcheck_email/healthcheck_email.py]: Please use `flask_restful`, this package was removed from the application `flask_restplus`

File: src/api/endpoints/healthcheck_email/healthcheck_email.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
+"""Module to test Notifications being sent out."""
```
Comment (georgreen) [inline src/api/endpoints/healthcheck_email/healthcheck_email.py]: If this module is for testing, doesn't it belong to the testing package then?

File: src/api/services/notifications/tasks.py [hunk 8]
```diff
@@ -1,7 +1,10 @@
 """Notifications module."""
 import os
-import requests
+# import requests
```
Comment (georgreen) [inline src/api/services/notifications/tasks.py]: You forgot this coment?

File: src/api/services/notifications/tasks.py [hunk 9]
```diff
@@ -47,37 +56,52 @@ def send_email(**kwargs):
...
+        msg = Message(subject=kwargs["subject"],
+                      recipients=kwargs["recipients"], sender=kwargs["sender"])
+        msg.body = kwargs["message"]
+
+        if kwargs.get("html"):
+            msg.html = kwargs.get("html")
+
+        mail_sender.send(msg)
+
+        # NOTE:  These will be used in future when the problems with MailGun
```
Comment (georgreen) [inline src/api/services/notifications/tasks.py]: Git can do a much better job than, leaving comments. You can safely delete this if required `git checkout <previous version>` will suffice to bring it back.

File: src/requirements.txt [hunk 10]
```diff
@@ -13,13 +15,16 @@ coveralls==1.2.0
...
+Flask==0.12.2
 Flask-Cors==3.0.3
+Flask-Mail==0.9.1
 Flask-Migrate==2.1.1
 Flask-RESTful==0.3.6
+flask-restplus==0.10.1
```
Comment (georgreen) [inline src/requirements.txt]: Remove this, please.

File: src/requirements.txt [hunk 11]
```diff
@@ -54,7 +59,11 @@ requests==2.18.4
 simplegeneric==0.8.1
 six==1.10.0
 SQLAlchemy==1.1.14
+tornado==5.1
 traitlets==4.3.2
 urllib3==1.22
 vine==1.1.4
+virtualenv==16.0.0
```
Comment (georgreen) [inline src/requirements.txt]: These are dev dependency and should not be here.

File: src/api/endpoints/healthcheck_email/healthcheck_email.py (hunk 7 above)
Comment (Andretalik) [reply src/api/endpoints/healthcheck_email/healthcheck_email.py]: Its an endpoint that can be hit to confirm if the notifications are running, Just like a health-check.

**Pull request merged**